src/bittybuzz/src/BittyBuzzMessageHandler.cpp

```diff
@@ -35,61 +35,112 @@ bool BittyBuzzMessageHandler::processMessage() {
+            m_logger.log(LogLevel::Warn, "BBZ: function call request args length don't match");
 
-        // Pushing bbz table for arguments
-        bbzheap_idx_t table = bbztable_new();
+            invalidRequestResponse.setDetails("arg length don't match");
+            return invalidRequestResponse;
+        }
 
+        bbzvm_push(registeredClosure.m_selfHeapIdx); // Push self table
+        bbzvm_push(registeredClosure.m_closureHeapIdx); // Push closure
+
+        // Pushing bbz args
         for (uint16_t i = 0; i < functionRequest.getArgumentsLength(); i++) {
             // Buzz table index
-            bbzheap_idx_t tableIdx = bbzint_new((int16_t)i);
-
             const std::variant<std::monostate, int64_t, float>& arg = args[i].getArgument();
 
             if (const int64_t* intVal = std::get_if<int64_t>(&arg)) {
-                bbzheap_idx_t bbzIntVal = bbzint_new(*intVal);
-                bbztable_set(table, tableIdx, bbzIntVal);
-            }
-
-            else if (const float* floatVal = std::get_if<float>(&arg)) {
-                bbzheap_idx_t bbzFloatVal = bbzfloat_new(bbzfloat_fromfloat(*floatVal));
-                bbztable_set(table, tableIdx, bbzFloatVal);
+                // Check if it matches the stored type
+                if (std::get<1>(storedArgs[i]) != FunctionDescriptionArgumentTypeDTO::Int) {
+                    invalidRequestResponse.setDetails("arg type don't match");
+                    return invalidRequestResponse;
+                }
+                bbzvm_pushi((int16_t)*intVal);
+
+            } else if (const float* floatVal = std::get_if<float>(&arg)) {
+                if (std::get<1>(storedArgs[i]) != FunctionDescriptionArgumentTypeDTO::Float) {
+                    invalidRequestResponse.setDetails("arg type don't match");
+                    return invalidRequestResponse;
+                }
+                bbzvm_pushf(bbzfloat_fromfloat(*floatVal));
             }
         }
 
-        // TODO: Migrate to variable number of arguments once we know the number beforehand (at
-        // registration), will avoid the user messing with tables (if possible)
-        bbzvm_pushnil(); // Push self table
-        bbzvm_push(closureHeapIdx.value());
-        bbzvm_push(table);
-        bbzvm_closure_call(1);
+        // Call the closure
+        bbzvm_closure_call(functionRequest.getArgumentsLength());
         bbzvm_pop(); // Pop self table
 
         // response
         return FunctionCallResponseDTO(GenericResponseStatusDTO::Ok, "");
     }
-    return FunctionCallResponseDTO(GenericResponseStatusDTO::BadRequest, "");
+    return FunctionCallResponseDTO(GenericResponseStatusDTO::BadRequest, "Not registred");
 }
 
 UserCallResponseDTO BittyBuzzMessageHandler::handleUserCallRequest(
     const UserCallRequestDTO& userRequest) {
 
-    const std::variant<std::monostate, FunctionCallRequestDTO>& variantReq =
-        userRequest.getRequest();
+    // TODO: handle
```

Comment:
to address or for later?

Reply:
Later, idk how we will handle that, probably a type of hashmap to store previous unresolved request, though that's a lot of code and planning. I don't even know if we will have enough memory to do this since this would require a buffer of messageDTO that may or may not need to be big

Reply:
Already done actully, forgot to remove it